dx: Dev bootstrap

nemanjaglumac opened the pull request:
## Summary

This PR improves the developer onboarding experience with two main additions:

1. New bin/dev-bootstrap script

A one-command setup for new developers that:
- Verifies required tools are installed (node, yarn, java, clojure)
- Installs frontend dependencies (`yarn install --frozen-lockfile --prefer-offline`)
- Downloads backend dependencies (most commonly used Clojure aliases)
- Optionally builds the Metabase JAR (with age display for existing JARs)

2. New bin/utils/ shared utilities

A modular library of bash utilities to reduce duplication across dev scripts:
- colors.sh - TTY-aware color definitions
- logging.sh - Standardized print functions (print_header, print_step, print_success, etc.)
- prompts.sh - Interactive prompts (ask_yes_no)
- files.sh - File utilities (get_file_age)
- all.sh - Convenience loader

3. Updates to bin/dev-requirements
- Now uses shared utilities
- Added optional tools: gh, op, aws, orb
- Improved formatting for optional tools list

### Test plan

- Run ./bin/dev-requirements and verify output formatting
- Run ./bin/dev-bootstrap on a fresh clone (or after rm -rf node_modules)
- Verify JAR age detection works when JAR exists
- Test on both macOS and Linux (file age uses platform-specific stat)

nemanjaglumac committed:
Update optional tooling in dev-requirements

nemanjaglumac committed:
Add dev-bootstrap

nemanjaglumac committed:
Separate the utilities

nemanjaglumac committed:
Improve instructions

nemanjaglumac committed:
Fix linter

nemanjaglumac committed:
Formatting

nemanjaglumac committed:
Add lock file to prevent multiple bootstraps

escherize commented on bin/dev-requirements:
```diff
@@ -334,44 +323,32 @@ main() {
+        echo "    - Clojure CLI"
     fi
-    echo "    • Babashka (for mage task automation)"
+    echo "    - Babashka (for mage task automation)"
```
it's semantics, but you don't need to install babashka to use mage. As it is now, the `:min-bb-version` of bb (from bb.edn) is auto-installed into `./bin/bb` and only that version is used by mage, not what is on people's path.

escherize commented on bin/dev-bootstrap:
```diff
@@ -0,0 +1,148 @@
+        echo ""
+        echo "Run ./bin/dev-requirements to check your development environment."
+        echo ""
+        echo "If you'd like an automated setup, run ./bin/dev-install (uses mise)."
```
not a big fan of referencing scripts that don't exist yet 😓

escherize reviewed (commented):
Looks good overall, I had some questions about how stuff actually works. Especially the 3033 port, first I've seen that.

nemanjaglumac commented on bin/dev-requirements:
```diff
@@ -334,44 +323,32 @@ main() {
+    print_optional_tool rg "ripgrep - A fast alternative to grep" "https://github.com/BurntSushi/ripgrep"
+    print_optional_tool gh "gh - GitHub CLI" "https://cli.github.com/"
+    print_optional_tool op "op - 1Password CLI" "https://1password.com/downloads/command-line/"
+    print_optional_tool aws "aws - AWS CLI" "https://aws.amazon.com/cli/"
```
mise will auto install all of this for you

nemanjaglumac committed:
Update clojure example

Co-authored-by: bryan <bryan.maass@gmail.com>

escherize commented on bin/dev-bootstrap:
```diff
@@ -0,0 +1,148 @@
+    fi
+
+    if ! command -v java &> /dev/null; then
+        missing+=("java")
```
if mise can guarantee that a version is installed, why not check that the version installed matches what's in `mise.toml`?

escherize commented on bin/utils/logging.sh:
```diff
@@ -0,0 +1,62 @@
+# Print a section header (for grouping related output)
+print_section() {
+    echo -e "\n${BOLD}$1${NC}"
+    echo -e "${BOLD}$(printf -- '-%.0s' {1..63})${NC}"
```
want to match the print_header function's length?

```suggestion
    echo -e "${BOLD}$(printf -- '-%.0s' {1..67})${NC}"
```

escherize committed:
minor cleanup

- add comment about skipping node/java/etc. version checks
- mention there's no such thing as bin/dev-install yet
- better app startup reccomendation
- remove port from java .. -jar  .. command -> default port by default
- make print_section length match print_header
